src/main/kotlin/exercise3/task4/TaxiParkTask.kt

```diff
@@ -10,29 +10,56 @@ package exercise3.task4
  * Find all the drivers who performed no trips.
  */
 internal fun TaxiPark.findFakeDrivers(): Set<Driver> {
-    TODO("Implement me!!!")
+    return allDrivers.filter { driver -> trips.none { it.driver == driver } }
```

Comment:
This is a correct solution, but its performance is not great because we'll have to do D * T iterations, where D is the number of drivers, and T is the number of trips.

The same solution can be done by mapping all the drivers from trips and then deducting all drivers from the list of drivers present in trips.

allDrivers - trips.map { it.driver }
This way we only once iterate over the trips list.